packages/editor/src/components/block-list/index.js

```diff
@@ -182,37 +190,54 @@ class BlockList extends Component {
 			blockClientIds,
 			rootClientId,
 			isDraggable,
+			selectedBlockClientId,
+			multiSelectedBlockClientIds,
 		} = this.props;
 
 		return (
 			<div className="editor-block-list__layout">
-				{ map( blockClientIds, ( clientId, blockIndex ) => (
-					<BlockListBlock
-						key={ 'block-' + clientId }
-						index={ blockIndex }
-						clientId={ clientId }
-						blockRef={ this.setBlockRef }
-						onSelectionStart={ this.onSelectionStart }
-						rootClientId={ rootClientId }
-						isFirst={ blockIndex === 0 }
-						isLast={ blockIndex === blockClientIds.length - 1 }
-						isDraggable={ isDraggable }
-					/>
-				) ) }
+				{ map( blockClientIds, ( clientId, blockIndex ) => {
+					return (
+						<AsyncModeProvider
+							key={ 'block-' + clientId }
+							value={
+								selectedBlockClientId !== clientId &&
+							( !! selectedBlockClientId || multiSelectedBlockClientIds.indexOf( clientId ) === -1 )
+							}
+						>
+							<BlockListBlock
+								clientId={ clientId }
+								index={ blockIndex }
+								blockRef={ this.setBlockRef }
+								onSelectionStart={ this.onSelectionStart }
+								rootClientId={ rootClientId }
+								isFirst={ blockIndex === 0 }
+								isLast={ blockIndex === blockClientIds.length - 1 }
+								isDraggable={ isDraggable }
+							/>
+						</AsyncModeProvider>
+					);
+				} ) }
 				<BlockListAppender rootClientId={ rootClientId } />
 			</div>
 		);
 	}
 }
 
 export default compose( [
+	// This component needs to always be synchronous
```

Comment:
Can you elaborate on why it needs to be wrapped with `AsyncModeProvider` component in this place? It sets async mode to `false` so I would expect it it equivalent to what we have as of today.

Reply:
Because in nested contexts, this component can be inside another block in async mode but this particular component should always be sync because it's responsible of switching the mode.

Reply:
> this component can be inside another block

Might be worth clarifying this point in the existing comment.